model/user/db.go

```diff
@@ -306,16 +307,34 @@ func GetOrCreateUser(userId, displayName, email, accessToken, refreshToken strin
 		DispNameKey:     displayName,
 		EmailAddressKey: email,
 	}
-	if accessToken != "" {
-		setFields[bsonutil.GetDottedKeyName(LoginCacheKey, LoginCacheAccessTokenKey)] = accessToken
-	}
 	if refreshToken != "" {
 		setFields[bsonutil.GetDottedKeyName(LoginCacheKey, LoginCacheRefreshTokenKey)] = refreshToken
 	}
 	setOnInsertFields := bson.M{
 		APIKeyKey: utility.RandomString(),
 		bsonutil.GetDottedKeyName(SettingsKey, UseSpruceOptionsKey, SpruceV1Key): true,
 	}
+
+	if accessToken != "" {
+		setFields[bsonutil.GetDottedKeyName(LoginCacheKey, LoginCacheAccessTokenKey)] = accessToken
+	} else {
+		// only fetch the slack user if the evergreen user doesn't exist (which is the case when there is no access token)
```

Comment:
The access token is an Okta-specific field, but I don't think it can be relied on to indicate that it's the first login ever because it could be cleared if the user is logged out. I imagine what we actually want is to check if the user we're about to update has `SlackMemberId`/`SlackUsername` set, regardless of whether it's their first or Nth login. Would it be okay to just look up the user, see if their Slack settings are populated (or if the user doesn't exist yet), and fill them in here if they're empty?

Reply:
wait, ignore me, lol

Reply:
updated.